### DISCUSSION TITLE
GitHubFlow workflow isn't bumping versions

### DISCUSSION THREAD

**Discussion by travaille-dev** [GitTools/GitVersion#4166] [Q&A] [answered]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

GitVersion.Tool

### GitVersion version

6.0.2

### Operating system

Linux, macOS

## Background
I have a class library. With our ci/cd pipeline for prs we test, build, and publish that nuget package to a snapshot nuget repo.
After a pr gets merged, we'd like to consider this a 'release', version accordingly, and publish to a production nuget repo. That means no pre release tags.

We use conventional commits to bump versions based on commit messages. Things have upgraded with the library since the docs have provided guidance on conventional commits and I don't think mainline mode is supported anymore

Our config
``` yml
workflow: GitHubFlow/v1
major-version-bump-message: "^(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test|config)(\\([\\w\\s-,/\\\\]*\\))?(!:|:.*\\n\\n((.+\\n)+\\n)?BREAKING CHANGE:\\s.+)"
minor-version-bump-message: "^(feat)(\\([\\w\\s-,/\\\\]*\\))?:"
patch-version-bump-message: "^(fix|perf|config|refactor|chore)(\\([\\w\\s-,/\\\\]*\\))?:"
ignore:
  sha: []
```

## Issue
I can't get the minor/patch to increment for my life on a feature branch.

I have tried removing my bump-message overrides to just the default `+semver: whatever` ones and still it's a no go.

### What is expected?

If I write a commit message like `chore(help):please bump the patch` I would like the patch to be bumped up. 
If I write a commit message like `feat(help): please bump the minor` I would like the. minor to be bumped up.

If I'm thinking of this workflow incorrectly then let me know, but how can I get these major/patch increments to work.
the full semver does increment because it has build metadata

### Steps to Reproduce

have 6.0.2 installed as a dotnet tool
create a feature/fake branch off of main
make commit(s) like `chore(help):please bump the patch` and `feat(help): please bump the minor` 
run `dotnet-gitversion` and see no minor or patch increments. Just the build metadata

**Comment by HHobeck** [comment]
Please only create a bug report if you are sure, you found a bug. Have you invested any time to find out the root cause by yourself or do you just created a issue after you faced a problem?

1. We have docent of automated integration tests which covers your scenario
2. We have documented use case scenarios which can be used as a reference
3. If you are changing the configuration you need to know what you are doing

Link[1]: [Create feature branch from develop](https://gitversion.net/docs/learn/branching-strategies/gitflow/examples)
![image](https://github.com/user-attachments/assets/8308c8a8-b739-4035-9c4b-cccfe655ba53)

Link[2]: https://regex101.com/
![image](https://github.com/user-attachments/assets/3b558cd5-74fe-48b7-a748-62f514ab42b3)

I would recommend you to start with the a simple workflow and use the default message bump regex before changing it.

**Reply by travaille-dev** [reply]
Sure. The regex was gotten from https://gitversion.net/docs/reference/version-increments with basic modification, but all right.
I can show you the example with just the defaults for githubflow and patch still does not increment.
```yml
workflow: GitHubFlow/v1
mode: ContinuousDeployment
ignore:
  sha: []
```

## Test one
This shows that when I make a new feature branch that the patch is not bumped up and patch is not incremented when I explicitly put `+semver: patch` in the commit message. However, it 

``` mermaid
gitGraph TB:
       commit id:" " tag: "semver: 1.1.3"
       branch feature/fake-stuff
       checkout feature/fake-stuff
       commit id:"+semver: patch" tag: "semver: 1.1.3-fake-stuff.1"
       commit id:"+semver: minor" tag: "semver: 1.2.0-fake-stuff.1"
       commit id:"+semver: patch" tag: "semver: 1.2.0-fake-stuff.1"
       commit id:"+semver: major" tag: "semver: 2.0.0-fake-stuff.1"
       commit id:"+semver: patch" tag: "semver: 2.0.0-fake-stuff.1"
```
## Test two
this just shows that patch is not incremented when I make a new feature branch
``` mermaid
gitGraph TB:
       commit id:" " tag: "semver: 1.1.3"
       branch feature/fake
       checkout feature/fake
       commit id:" " tag: "semver: 1.1.3-fake.1"
```

## Test three
this shows patch not incrementing when i make an unknown branch
``` mermaid
gitGraph TB:
       commit id:" " tag: "semver: 1.1.3"
       branch asdfasdf
       checkout asdfasdf
       commit id:"+semver: patch" tag: "semver: 1.1.3-asdfasdf.1"
       commit id:"+semver: fix" tag: "semver: 1.1.3-asdfasdf.1"
       commit id:"+semver: patch" tag: "semver: 1.1.3-asdfasdf.1"
       commit id:"+semver: minor" tag: "semver: 1.2.0-asdfasdf.1"
```


You guys don't really have test cases for the `+semver: fix/patch` commits so I'm trying to get an understanding if this is as designed. It does look like from the following that I would have my feature branches bumped up automatically.
src https://gitversion.net/docs/learn/branching-strategies/githubflow/examples
![](https://gitversion.net/docs/img/DocumentationSamplesForGitHubFlow_FeatureBranch.png)

**Reply by HHobeck** [reply]
Test one:
Please take a look to the following discussion to get an understanding how the bump messages are working:
https://github.com/GitTools/GitVersion/discussions/3933#discussioncomment-8600888

Test two:
The version is incremented based on the last tag. I guess your last tag on main is 1.1.2. The increment is set to inherit one the feature branch... That means git version takes the increment from main which is patch.

Test three:
Is the same like test two. GitVersion calculates the next version based on the last tag or 0.0.0 if no tag is existent. The increment is set to inherit one the unknown branch... That means git version takes the increment from main which is patch.

> You guys don't really have test cases for the +semver: fix/patch commits so I'm trying to get an understanding if this is as designed. It does look like from the following that I would have my feature branches bumped up automatically.

If you define the increment to e.g. minor on the feature branch configuration then the minor version will be incremented (once) until you are releasing your software to proudction (RTM) by tagging the version number on main.

Edit: The expression `+semver: patch` makes only sense if you have a branch with increment `none` which is not the case in the GitHubFlow workflow.

**Comment by travaille-dev** [comment, marked as answer]
Yeah I think it's just the tags I was missing. I didn't want to have to add a tag until I actually went in and made a release in github.
However, it's probably more performant to just look at the last tag on the main branch rather than parsing out all the git commit messages from all time.

So I've adjusted my build automation to push a tag out after merging to main.

As for bumping patch versions I can just change my workflow around. I was expecting to be able to bump the patch multiple times while working on a feature branch, but honestly one patch bump when I branch off of main makes sense and I can live with it.
